tests/core/test_initialization.py

```diff
@@ -273,83 +273,76 @@ def setUp(self):
         self.config.registry.cache = {}
         self.config.registry.permission = {}
 
-    @mock.patch('kinto.core.statsd.Client')
-    def test_statsd_isnt_called_if_statsd_url_is_not_set(self, mocked):
+        patch = mock.patch('kinto.core.statsd.load_from_config')
+        self.mocked = patch.start()
+        self.addCleanup(patch.stop)
+
+    def test_statsd_isnt_called_if_statsd_url_is_not_set(self):
         self.config.add_settings({
             'statsd_url': None
         })
         initialization.setup_statsd(self.config)
-        mocked.assert_not_called()
+        self.mocked.assert_not_called()
 
-    @mock.patch('kinto.core.statsd.Client')
-    def test_statsd_is_set_to_none_if_statsd_url_not_set(self, mocked):
+    def test_statsd_is_set_to_none_if_statsd_url_not_set(self):
         self.config.add_settings({
             'statsd_url': None
         })
         initialization.setup_statsd(self.config)
         self.assertEqual(self.config.registry.statsd, None)
 
-    @mock.patch('kinto.core.statsd.Client')
-    def test_statsd_is_called_if_statsd_url_is_set(self, mocked):
+    def test_statsd_is_called_if_statsd_url_is_set(self):
         initialization.setup_statsd(self.config)
-        mocked.assert_called_with('host', 8080, 'kinto.core')
+        self.mocked.assert_called_with(self.config)
+        # See `tests/core/test_statsd.py` for instantiation tests.
 
-    @mock.patch('kinto.core.statsd.Client')
-    def test_statsd_is_expose_in_the_registry_if_url_is_set(self, mocked):
+    def test_statsd_is_expose_in_the_registry_if_url_is_set(self):
         initialization.setup_statsd(self.config)
-        self.assertEqual(self.config.registry.statsd, mocked.return_value)
+        self.assertEqual(self.config.registry.statsd, self.mocked.return_value)
```

Comment:
i'm not saying you should use that (it depends on wether you want conciseness or code coherence), but just so you know, with pytest you can use simple `assert`s

```assert self.config.registry.statsd == self.mocked.return_value```